src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/boss_rage_winterchill.cpp

```diff
@@ -15,156 +15,160 @@
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+/*
+ * Combat timers requires to be revisited
+ */
+
 #include "ScriptMgr.h"
 #include "hyjal.h"
-#include "hyjal_trash.h"
-#include "InstanceScript.h"
-#include "ObjectAccessor.h"
+#include "MotionMaster.h"
+#include "SpellInfo.h"
+
+enum WinterchillTexts
+{
+    SAY_INTRO                   = 0,
+    SAY_DECAY                   = 1,
+    SAY_NOVA                    = 2,
+    SAY_SLAY                    = 3,
+    SAY_DEATH                   = 4
+};
 
-enum Spells
+enum WinterchillSpells
 {
+    SPELL_ICEBOLT               = 31249,
+    SPELL_FROST_NOVA            = 31250,
     SPELL_FROST_ARMOR           = 31256,
     SPELL_DEATH_AND_DECAY       = 31258,
-    SPELL_FROST_NOVA            = 31250,
-    SPELL_ICEBOLT               = 31249
+    SPELL_BERSERK               = 26662
 };
 
-enum Texts
+enum WinterchillEvents
 {
-    SAY_ONDEATH                 = 0,
-    SAY_ONSLAY                  = 1,
-    SAY_DECAY                   = 2,
-    SAY_NOVA                    = 3,
-    SAY_ONAGGRO                 = 4
+    EVENT_ICEBOLT               = 1,
+    EVENT_FROST_NOVA,
+    EVENT_FROST_ARMOR,
+    EVENT_DEATH_AND_DECAY,
+    EVENT_BERSERK
 };
 
-class boss_rage_winterchill : public CreatureScript
+// 17767 - Rage Winterchill
+struct boss_rage_winterchill : public BossAI
 {
-public:
-    boss_rage_winterchill() : CreatureScript("boss_rage_winterchill") { }
+    boss_rage_winterchill(Creature* creature) : BossAI(creature, DATA_RAGEWINTERCHILL) { }
 
-    CreatureAI* GetAI(Creature* creature) const override
+    void JustAppeared() override
     {
-        return GetHyjalAI<boss_rage_winterchillAI>(creature);
+        Talk(SAY_INTRO);
+        me->setActive(true);
```

Comment:
i understand the creature won't update and start walking without this (we actually set entities using BossAI active during an encounter in progress), but this is bad, and as far as I recall, in order for this to work properly without forcing active is allowing all grids to update within an instance (there's a config for this)

we had the same issue in Silverpine where the transport moving across the whole map wouldn't update without active being set and it was rejected since the way of simulating it is through the config

Reply:
Thing is even without this all summoned creatures will be active and far visible (grid size), it's done in separate file
I'm just duplicating it so this file will contain everything that is needed and we can forgot about bosses in next updates